api/desecapi/views.py

```diff
@@ -120,6 +122,95 @@ class DomainDetailByName(DomainDetail):
+    lookup_field = 'subname'
+    serializer_class = RRsetSerializer
+    permission_classes = (permissions.IsAuthenticated, IsDomainOwner,)
+
+    def get_queryset(self):
+        name = self.kwargs['name']
+        subname = self.kwargs['subname']
+        type = self.kwargs['type']
+
+        return RRset.objects.filter(domain__name=name, subname=subname, type=type)
+
+    def delete(self, request, *args, **kwargs):
+        try:
+            return super().delete(request, *args, **kwargs)
+        except django.core.exceptions.ValidationError as e:
+            ex = ValidationError(detail={"detail": str(e)})
+            ex.status_code = status.HTTP_409_CONFLICT
+            raise ex
+
+    def get(self, request, *args, **kwargs):
+        subname = self.kwargs['subname']
+        type = self.kwargs['type'].upper()
+
+        # TODO Do we need this functionality at all? We could just disregard the SOA record completely.
+        if type == 'SOA' and subname == '':
+            name = self.kwargs['name']
+
+            # This will throw an exception if the user does not own the domain
+            try:
+                domain = Domain.objects.get(name=name, owner=request.user.pk)
+            except Domain.DoesNotExist:
+                raise Http404
+
+            return Response(pdns.get_rrsets(domain.name, subname, type)[0])
+
+        return super().get(request, *args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        records = request.data.get('records')
+
+        if records == []:
+            return self.delete(request, *args, **kwargs)
+
+        try:
+            return super().update(request, *args, **kwargs)
+        except django.core.exceptions.ValidationError as e:
+            ex = ValidationError(detail={"detail": str(e)})
+            ex.status_code = status.HTTP_409_CONFLICT
+            raise ex
+        # TODO add good exception for PUT/PATCH'ing the SOA record, if we decide to take of SOA at all
+
+
+class RRsetsDetail(generics.ListCreateAPIView):
+    serializer_class = RRsetSerializer
+    permission_classes = (permissions.IsAuthenticated, IsDomainOwner,)
+
+    def get_queryset(self):
+        name = self.kwargs['name']
+        subname = self.kwargs.get('subname')
+        type = self.kwargs.get('type')
+
+        # Replace by proper exception
+        assert(subname is None or type is None)
+
+        if subname is not None:
+            return RRset.objects.filter(domain__name=name, domain__owner=self.request.user.pk, subname=subname)
+        elif type is not None:
+            return RRset.objects.filter(domain__name=name, domain__owner=self.request.user.pk, type=type)
+        else:
+            return RRset.objects.filter(domain__name=name, domain__owner=self.request.user.pk)
+
+    def create(self, request, *args, **kwargs):
+        if self.kwargs.get('subname') is not None or self.kwargs.get('type') is not None:
+            raise MethodNotAllowed(method=request.method)
```

Comment:
Shouldn't this be `CONFLICT`? Method not allowed tells IMHO that this method is not allowed at this endpoint under any circumstances.